src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -20800,6 +20799,62 @@ static void F(B? b)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "b").WithArguments("a", "A.explicit operator S(A a)").WithLocation(13, 17));
         }
 
+        // PROTOTYPE(NullableReferenceTypes): Should only report one warning for: A a = (B)s;
+        [Fact]
+        public void MultipleConversions_Explicit_04()
+        {
+            var source =
+@"struct S
+{
+    public static explicit operator A?(S s) => throw null;
+}
+class A
+{
+    internal void F() { }
+}
+class B : A
+{
+}
+class C
+{
+    static void F(S s)
+    {
+        var b1 = (B)s;
+        b1.F();
+        B b2 = (B)s;
+        b2.F();
+        A a = (B)s;
+        a.F();
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (16,18): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         var b1 = (B)s;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(16, 18),
+                // (17,9): warning CS8602: Possible dereference of a null reference.
+                //         b1.F();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b1").WithLocation(17, 9),
+                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         B b2 = (B)s;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
+                // (18,16): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         B b2 = (B)s;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(B)s").WithLocation(18, 16),
```

Comment:
>WRN_ConvertingNullableToNonNullable [](start = 37, length = 35)

There seems to be duplicate diagnostics (on assignment to `b2` and `a`)

Reply:
Yes, noted in `// PROTOTYPE` comment above.

---
In reply to: [189129732](https://github.com/dotnet/roslyn/pull/26356#discussion_r189129732) [](ancestors = 189129732)